### ISSUE TITLE
queryRenderedFeatures loses Feature IDs when using clustering

### ISSUE THREAD

**Issue by SamiPatanen** [mapbox/mapbox-gl-native#12347]
Hi!

I'm using queryRenderedFeatures to check what features are tapped by the user. This works great, until I set `new GeoJsonOptions().withCluster(true)` to the source of the layer I'm querying. I'm not expecting queryRenderedFeatures to work for individual features when the features are clustered, only when they are displayed separately when zoomed in.

### Steps to reproduce

 1. Create a layer with clustering set to true and add a bunch of features with IDs to that layer.
 2. Use queryRenderedFeatures for that layer, with camera zoomed in close enough that the features are not clustered.
 3. Check the returned features for IDs.

### Expected behavior

The IDs are present, as without clustering.

### Actual behavior

The ID is an empty string.

### Configuration

**Android versions:** 8.1.0
**Device models:** Nexus 5X
**Mapbox SDK versions:** Tested on 5.5.0-5.5.3 and 6.2.1.

**Comment by tobrun** [comment]
@SamiPatanen the library responsible for providing clustering is the supercluster.hpp project [here](https://github.com/mapbox/supercluster.hpp/). Quickly glanced through the code and seeing constructions as [these](https://github.com/mapbox/supercluster.hpp/blob/master/include/supercluster.hpp#L135). This means that the clustering algorithm overwrites/replaces feature ids. To unblock your use-case, you can  create a new id-like field and add/use that one instead. 

For this issue, if you feel this should be addressed, can you create a new ticket upstream in supercluster.hpp? We won't be able to fix this up in codebase found in this repository.

Thank you for reaching out and reporting this to us!
(closing as this should be reopened upstream)

**Comment by SamiPatanen** [comment]
Hi @tobrun!

"To unblock your use-case, you can create a new id-like field and add/use that one instead."

I assume you mean something like:
```
String id = "something";
feature.addStringProperty("customID", id);
```

And then retrieving this property after queryRenderedFeatures like so:
```
String id = feature.getStringProperty("customID");
```

This results in a NullPointerException:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.gson.JsonElement.getAsString()' on a null object reference
        at com.mapbox.geojson.Feature.getStringProperty(Feature.java:324)
```

**Comment by tobrun** [comment]
@SamePatanen, wasn't expecting this behaviour tbh, can you open a ticket in the supercluster repository? thanks!

**Comment by SamiPatanen** [comment]
@tobrun I opened an issue in the supercluster repository, and it was quickly closed, as they were already working on improving related things. However, they said that the supercluster shouldn't remove features.

I tested without clustering, and it resulted in a NullPointerException as well. It turns out I managed to mess it up somehow. So the workaround works well after all.
